compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1339,6 +1343,13 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
           cls, isRequired, cdef.pos)
     }
 
+    if (cls.classParents.exists(_.classSymbol eq defn.PhantomClass)) {
```

Comment:
To make sure that all phantom type prefixes are static. This restrictions could be relaxed latter on.

If we don have this restriction, what would be the semantics of the following code:
```scala
class A extends Phantom {
  def myPhantom: this.Any = assume 
}
class B extends A
```
What would be the relation between `A.Any` and `B.Any`?